aten/src/ATen/native/cuda/UpSampleNearest1d.cu

```diff
@@ -109,6 +109,15 @@ static void upsample_nearest1d_out_cuda_template(
   dim3 bdim{std::min<unsigned int>(
       at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock, MAX_THREADS)};
   dim3 gdim{cuda::ATenCeilDiv(n, bdim.x)};
+  // Error out on cases where gdim.x exceeds limit of launch config, as
+  // the current kernel implementation doesn't loop over non-batch dimensions.
+  // Note that it is unlikely to happen, as maxGridSize[0] is 2**31-1
+  // TODO: kernel implementation could stride on spatial dimension. We probably
+  //       need to overhaul the kernel.
+  TORCH_CHECK(
+      gdim.x <= at::cuda::getCurrentDeviceProperties()->maxGridSize[0],
+      "input tensor has spatial dimension larger than the kernel capacity");
+
```

Comment:
tensors are indexed with ints in the kernel, so just check that the number of elements fits, and don't do this check.

Reply:
It's fair to assume that maxGridSize is not going to change in the future.
I'll update it for the places where it's the same kernel launch.